### PR TITLE
docs: channel upgrade timeout

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #5268 

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (colin-axner) [inline docs/docs/01-ibc/06-channel-upgrades.md:152]:
```suggestion
Timing out an outstanding channel upgrade may be necessary during the flushing packet stage of the channel upgrade process. As stated above, with `ChanUpgradeTry` or `ChanUpgradeAck`, the channel state has been changed from `OPEN` to `FLUSHING`, so no new packets will be allowed to be sent over this channel while flushing. If upgrades cannot be performed in a timely manner (due to unforeseen flushing issues), upgrade timeouts allow the channel to avoid blocking packet sends indefinitely. If flushing exceeds the time limit set in the `UpgradeTimeout` channel `Params`, the upgrade process will need to be timed out to abort the upgrade attempt and resume normal channel processing. 
```

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (colin-axner) [inline docs/docs/01-ibc/06-channel-upgrades.md:154]:
```suggestion
Channel upgrades require setting a valid timeout value in the channel `Params` before submitting a `MsgChannelUpgradeTry` or `MsgChannelUpgradeAck` message (by default, 10 minutes): 
```

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (colin-axner) [inline docs/docs/01-ibc/06-channel-upgrades.md:183]: I think the upgrade needs to be re-approved by governance if it times out? Can we note that the upgrade process starts from the beginning?

**Review by colin-axner — APPROVED** [review]
Thanks @charleenfei :pray:

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (DimitrisJim) [inline docs/docs/01-ibc/06-channel-upgrades.md:77]: might make sense to mention #5548 here?

**Review by DimitrisJim — APPROVED** [review]
looks great!

**Review by crodriguezvega — APPROVED** [review]
Nice work, @charleenfei! I think we should mention that height-based timeouts are not supported at the moment.

**Review by damiannolan — APPROVED** [review]

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (charleenfei) [reply docs/docs/01-ibc/06-channel-upgrades.md:171]: ah yes, ill update that, good catch!

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (charleenfei) [reply docs/docs/01-ibc/06-channel-upgrades.md:77]: good call, will mention it!

Comment (charleenfei) [reply docs/docs/01-ibc/06-channel-upgrades.md:77]: actually, ill update it after @chatton merges his docs so i can refer to those.

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (chatton) [inline docs/docs/01-ibc/06-channel-upgrades.md:69]:
```suggestion
As part of the handling of the `MsgChannelUpgradeInit` message, the application's `OnChanUpgradeInit` callback will be triggered as well.
```

**Review by chatton — APPROVED** [review]
LGTM! Just left one suggestion for re-wording

File: docs/docs/01-ibc/06-channel-upgrades.md
Comment (crodriguezvega) [reply docs/docs/01-ibc/06-channel-upgrades.md]: I added it [here](https://github.com/cosmos/ibc-go/pull/5546/commits/4e7129f70582bd1bdfc9b7d50b6b1bace996c44a).

Comment (DimitrisJim) [reply docs/docs/01-ibc/06-channel-upgrades.md]: stunning!

**Pull request merged**
